src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -193,14 +192,14 @@ func onBootstrapControlPlane(s *state.State, initConfig map[string]string) error
 	clusterConfig.Certificates.CACert = string(caPair.CertPem)
 	clusterConfig.Certificates.CAKey = string(caPair.KeyPem)
 
-	clusterConfig, err = clusterconfigs.Merge(clusterConfig, utils.ConvertBootstrapToClusterConfig(bootstrapConfig))
+	clusterConfig, err = types.MergeClusterConfig(clusterConfig, types.ClusterConfigFromBootstrapConfig(bootstrapConfig))
```

Comment:
This reads a bit weird. 
Does it make sense to have a `types/clusterconfigs` module?